rds.go

```diff
@@ -138,6 +162,33 @@ func (d *DedicatedDBAdapter) CreateDB(i *Instance, password string) (DBInstanceS
 	}
 }
 
+func (d *DedicatedDBAdapter) GetDBStatus(i *Instance) (DatabaseStatus, error) {
+	svc := rds.New(&aws.Config{Region: "eu-west-1"})
+	request := &rds.DescribeDBInstancesInput{
+		DBInstanceIdentifier: &i.Database,
+	}
+	result, err := svc.DescribeDBInstances(request)
+	status := DatabaseStatus{
+		State: "in progress",
+		Description: "Failed to retrieve state",
+	}
+	if yes := d.DidAwsCallSucceed(err); yes && len(result.DBInstances) > 0 {
+		databaseInstance := result.DBInstances[0]
+		switch *(databaseInstance.DBInstanceStatus) {
+		case "failed":
```

Comment:
are you planning on failed and incompatiable-parameters on the same case? go doesn't allow fallthough unless explicitly stated.
if so: `case "incompatible-parameters", "succeeded"`

source: https://github.com/golang/go/wiki/Switch